app/pages/viewuser/ui.test.js

```diff
@@ -0,0 +1,201 @@
+  await pageSetup(t, true);
+  await t.navigateTo(pageUrl);
+
+  const userNameHeading = Selector('h2[data-test-id="user-name-heading"]');
+  const userName = Selector('div[data-test-id="user-name"]');
+
+  await t
+    .expect(userNameHeading.exists).ok()
+    .expect(await extractInnerText(userNameHeading)).eql(content.nameHeading)
+    .expect(userName.exists).ok()
+    .expect(await extractInnerText(userName)).eql(userData.name);
+});
+
+test('should render contact details heading', async (t) => {
+  await pageSetup(t, true);
+  await t.navigateTo(pageUrl);
+
+  const contactDetailsHeading = Selector('h2[data-test-id="user-contact-details-heading"]');
+  const contactDetails = Selector('div[data-test-id="user-contact-details"]');
+
+  await t
+    .expect(contactDetailsHeading.exists).ok()
+    .expect(await extractInnerText(contactDetailsHeading)).eql(content.phoneNumberHeading)
+    .expect(contactDetailsHeading.exists).ok()
+    .expect(await extractInnerText(contactDetails)).eql(userData.phonenumber);
+});
+
+test('should render email address heading', async (t) => {
+  await pageSetup(t, true);
+  await t.navigateTo(pageUrl);
+
+  const emailHeading = Selector('h2[data-test-id="user-email-heading"]');
+  const email = Selector('div[data-test-id="user-email"]');
+
+  await t
+    .expect(emailHeading.exists).ok()
+    .expect(await extractInnerText(emailHeading)).eql(content.emailAddressHeading)
+    .expect(email.exists).ok()
+    .expect(await extractInnerText(email)).eql(userData.emailaddress);
+});
+
+test('should render edit user button', async (t) => {
+  await pageSetup(t, true);
+  await t.navigateTo(pageUrl);
+
+  const editUserButton = Selector('[data-test-id="edit-user-button"] a');
+
+  await t
+    .expect(editUserButton.exists).ok()
+    .expect(await extractInnerText(editUserButton)).eql(content.editUserButtonText)
+    .expect(editUserButton.hasClass('nhsuk-u-margin-bottom-9')).ok()
+    .expect(editUserButton.getAttribute('href')).eql('#');
+});
+
+test('should navigate to edit user page when edit user button is clicked', async (t) => {
+  await pageSetup(t, true);
+  await t.navigateTo(pageUrl);
+
+  const editUserButton = Selector('[data-test-id="edit-user-button"] a');
+
+  await t
+    .expect(editUserButton.exists).ok()
+    .click(editUserButton)
+    .expect(getLocation()).eql(`${pageUrl}#`);
+});
+
+test('should render disable account button', async (t) => {
+  await pageSetup(t, true);
+  await t.navigateTo(pageUrl);
+
+  const disableAccountButton = Selector('[data-test-id="disable-account-button"] a');
+
+  await t
+    .expect(disableAccountButton.exists).ok()
+    .expect(await extractInnerText(disableAccountButton)).eql(content.disableAccountButtonText)
+    .expect(disableAccountButton.hasClass('nhsuk-u-margin-bottom-9')).ok()
+    .expect(disableAccountButton.getAttribute('href')).eql('#');
+});
+
+test('should call the api when edit disable account button is clicked', async (t) => {
```

Comment:
should this be should navigate to ... rather than call the api?

Reply:
well we need to discuss this for the next story as I am not sure what this button will do - it will not be on a separate page I am guessins

Reply:
actually there is a confirmation page - changed description